### PR TITLE
Add createRecordValue API with BMap as a parameter

### LINKED ISSUE #28831: createRecordValue runtime api needs to have BMap has a param
**Description:**
ATM we have   createRecordValue(Module packageId, String recordTypeName,   Map<String, Object> valueMap) 

 So users create a map from their code side and pass that Map(ex. HashMap), then we iterate through it and create another BMap object. If we give an API to directly pass a BMap, then we can skip the iteration of creating BMap from java Map.

### PR DESCRIPTION
**Pull request by Nadeeshan96**
$title
Fixes #28831

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java [hunk 1]
```diff
@@ -830,10 +829,45 @@ public static BMap<BString, Object> createRecordValue(Module packageId, String r
     public static BMap<BString, Object> createReadonlyRecordValue(Module packageId, String recordTypeName,
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java:829]: Shall we deprecate other createReadonlyRecordValue takes Map as a parameter.

Comment (Nadeeshan96) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java:829]: Discussed not to deprecate these.

**Review by warunalakshitha — DISMISSED** [review]
LGTM

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java [hunk 2]
```diff
@@ -830,10 +829,45 @@ public static BMap<BString, Object> createRecordValue(Module packageId, String r
...
-        MapValueImpl<BString, Object> bmap = (MapValueImpl<BString, Object>) ValueUtils.createRecordValue(
+        MapValueImpl<BString, Object> bMapValue = (MapValueImpl<BString, Object>) ValueUtils.createRecordValue(
+                packageId, recordTypeName, valueMap);
+        bMapValue.freezeDirect();
+        return bMapValue;
+    }
+
+    /**
+     * Create a record value that populates record fields using the given package ID, record type name and
+     * a {@link BMap} of field names and associated values for fields.
```
Comment (HindujaB) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java:840]:
```suggestion
     * a {@link BMap} of field names and associated values for the fields.
```

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java [hunk 3]
```diff
@@ -830,10 +829,45 @@ public static BMap<BString, Object> createRecordValue(Module packageId, String r
...
+     *
+     * @param packageId      the package ID that the record type resides.
+     * @param recordTypeName name of the record type.
+     * @param valueMap       {@link BMap} of fields and values to be used when creating the record.
+     * @return value of the populated record.
+     * @throws BError if given record type is not defined in the ballerina module.
```
Comment (HindujaB) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java:846]:
```suggestion
     * @param packageId      the package ID where the record type is defined.
     * @param recordTypeName name of the record type.
     * @param valueMap       {@link BMap} of fields and values to initialize the record.
     * @return value of the populated record.
     * @throws BError if given record type is not defined in the ballerina module.
```

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java [hunk 4]
```diff
@@ -830,10 +829,45 @@ public static BMap<BString, Object> createRecordValue(Module packageId, String r
...
+                packageId, recordTypeName, valueMap);
+        bMapValue.freezeDirect();
+        return bMapValue;
+    }
+
+    /**
+     * Create a record value that populates record fields using the given package ID, record type name and
+     * a {@link BMap} of field names and associated values for fields.
+     *
+     * @param packageId      the package ID that the record type resides.
```
Comment (HindujaB) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java:842]: Shall we re-align the tabs in the doc comment?
The same applies to the other comments as well.

Comment (Nadeeshan96) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ValueCreator.java]: Addressed with bc568b8e63232c7927def3d91ae084d19f470bb7.

**Pull request merged**